src/Morphic-Base/MorphicWindowManager.class.st

```diff
@@ -0,0 +1,200 @@
+"
+I manage the native window, when i was create i open a window with a taskbar, if you create a window, the window is add at the taskbar.
+
+
+    Instance Variables
+	allWindow:		<Object>
+	allWindowState:		<Object>
+	allWorld:		<Object>
+	taskBar:		<Object>
+	taskBarWindow:		<Object>
+
+
+    Implementation Points
+"
+Class {
+	#name : #MorphicWindowManager,
+	#superclass : #Morph,
+	#instVars : [
+		'taskBar',
+		'taskBarWindow',
+		'rootSave',
+		'allAttributes',
+		'windows'
+	],
+	#category : #'Morphic-Base'
+}
+
+{ #category : #accessing }
+MorphicWindowManager >> allAttributes [
+	^ allAttributes
+]
+
+{ #category : #accessing }
+MorphicWindowManager >> allSaveRoot [
+	
+	^ rootSave 
+]
+
+{ #category : #accessing }
+MorphicWindowManager >> allWindow [ 
+
+	^ windows 
+]
+
+{ #category : #accessing }
+MorphicWindowManager >> closeAllWindowToLeft: aWindow [
+
+	| index |
+	index := windows indexOf: aWindow.
+	index - 1 to: 1 by: -1 do: [ :m | (windows at: m) delete ].
+	taskBar updateTasks
+]
+
+{ #category : #accessing }
+MorphicWindowManager >> closeAllWindowToRight: aWindow [
+
+	| index |
+	index := windows indexOf: aWindow.
+	windows size to: index + 1 by: -1 do: [ :m | 
+	(windows at: m) delete ].
+	taskBar updateTasks
+]
+
+{ #category : #accessing }
+MorphicWindowManager >> closeAllWindows [
+
+	
+	(windows size) to: (1) by: (-1) do: [ :m | (windows at: m) delete  ].
+	taskBar updateTasks .
+]
+
+{ #category : #accessing }
+MorphicWindowManager >> closeWindow: aWindow [
+
+	(windows includes: aWindow) ifTrue: [ windows remove: aWindow ].
```

Comment:
Could it happen that we are asked to close a window that we don't have?

Reply:
they do nothing, I add the comment on the method